File: scripts/fetch.js
// scripts/fetch.js

const https = require("https");
const fs = require("fs");

const URL = "https://comicw.net/map/";

function fetchPage() {
  return new Promise((resolve) => {
    https.get(URL, (res) => {
      let data = "";
      res.on("data", (chunk) => (data += chunk));
      res.on("end", () => resolve(data));
    });
  });
}

function extractVar(html, varName) {
  const regex = new RegExp(`const ${varName} = ([\\s\\S]*?);`);
  const match = html.match(regex);

  if (!match) {
    console.log(`${varName} 못찾음`);
    return {};
  }

  try {
    // Function으로 안전 실행
    const fn = new Function(`return ${match[1]}`);
    return fn();
  } catch (e) {
    console.log(`${varName} 파싱 실패`, e);
    return {};
  }
}

(async () => {
  const html = await fetchPage();

  const sat = extractVar(html, "APP_MAP_SAT");
  const sun = extractVar(html, "APP_MAP_SUN");

  if (!sat || !sun) {
    console.log("데이터 없음 → 종료");
    process.exit(1);
  }

  fs.mkdirSync("data/raw/latest", { recursive: true });

  fs.writeFileSync(
    "data/raw/latest/sat.json",
    JSON.stringify(sat, null, 2)
  );

  fs.writeFileSync(
    "data/raw/latest/sun.json",
    JSON.stringify(sun, null, 2)
  );

  console.log("fetch 완료");
})();

File: scripts/fetch.py
import requests
import re
import json
import os
import ast

URL = "https://comicw.net/map/"
res = requests.get(URL)
html = res.text


def extract(name):
    pattern = rf"const {name} = ({{.*?}});"
    match = re.search(pattern, html, re.S)

    if not match:
        print(f"{name} 못찾음")
        return {}

    text = match.group(1)

    # -------------------------
    # 🔥 JS → Python 변환 핵심
    # -------------------------

    # 1. 키 따옴표 추가
    text = re.sub(r"(\w+):", r'"\1":', text)

    # 2. 작은따옴표 → 큰따옴표
    text = text.replace("'", '"')

    # 3. trailing comma 제거
    text = re.sub(r",\s*}", "}", text)
    text = re.sub(r",\s*]", "]", text)

    try:
        return json.loads(text)
    except:
        print("json 실패 → ast 시도")

        try:
            return ast.literal_eval(text)
        except Exception as e:
            print("파싱 실패:", e)
            return {}


sat = extract("APP_MAP_SAT")
sun = extract("APP_MAP_SUN")


# -------------------------
# 저장
# -------------------------
os.makedirs("data/raw/latest", exist_ok=True)

with open("data/raw/latest/sat.json", "w", encoding="utf-8") as f:
    json.dump(sat, f, ensure_ascii=False, indent=2)

with open("data/raw/latest/sun.json", "w", encoding="utf-8") as f:
    json.dump(sun, f, ensure_ascii=False, indent=2)

print("fetch 완료")